comp/dogstatsd/server/server.go

```diff
@@ -373,8 +380,20 @@ func (s *server) start(context.Context) error {
 		}
 	}
 
-	if len(socketPath) > 0 {
-		unixListener, err := listeners.NewUDSDatagramListener(packetsChannel, sharedPacketPoolManager, sharedUDSOobPoolManager, s.config, s.tCapture, s.wmeta, s.pidMap)
+	// Listen on the default socket (if /var/run/datadog/ exists)
+	if _, err := os.Stat(filepath.Dir(defaultSocket)); !os.IsNotExist(err) {
```

Comment:
Would it make sense to have an option to disable this in the odd case it causes issues somewhere?

Reply:
Any reason to hide it? I think if there are problems, it would be best if users can discover and enable solution on their own, without hitting support.

Reply:
I discussed this with the team and the preferred option is to keep the exposed configuration for the client as simple as possible. Unless there is a known issue that opening this socket can cause, we can probably leave it as is. The code already checks for the existence of the parent directory, and should be harmless.

What do you think?

Reply:
I agree with @vickenty and it would be good to have a way to disable this feature completely as we never know what customers have done to secure their OS or may object to us enabling this on their system by default.

Reply:
Okay, I think I figured out a good middleground between overloading the config and doing uncontrollable things: Let's use the existing `dogstatsd_socket` and update the default value from `""` (disabled) to `/var/run/datadog/dsd.socket`, but only on non-windows systems. I see that this is a pattern already used in other places of the config template (for instance [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml#L1672-L1684) or [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml#L1841-L1869)).

Does this sound good from statsd perspective?